config/config.go

```diff
@@ -60,13 +60,22 @@ type (
 
 	// ApplicationConfiguration config specific to the node.
 	ApplicationConfiguration struct {
-		DataDirectoryPath string        `yaml:"DataDirectoryPath"`
-		RPCPort           uint16        `yaml:"RPCPort"`
-		NodePort          uint16        `yaml:"NodePort"`
-		Relay             bool          `yaml:"Relay"`
-		DialTimeout       time.Duration `yaml:"DialTimeout"`
-		ProtoTickInterval time.Duration `yaml:"ProtoTickInterval"`
-		MaxPeers          int           `yaml:"MaxPeers"`
+		DBConfiguration   DBConfiguration `yaml:"DBConfiguration"`
+		RPCPort           uint16          `yaml:"RPCPort"`
+		NodePort          uint16          `yaml:"NodePort"`
+		Relay             bool            `yaml:"Relay"`
+		DialTimeout       time.Duration   `yaml:"DialTimeout"`
+		ProtoTickInterval time.Duration   `yaml:"ProtoTickInterval"`
+		MaxPeers          int             `yaml:"MaxPeers"`
+	}
+
+	// DBConfiguration config for the DB used for storing BlockChain.
+	DBConfiguration struct {
+		DBType            string `yaml:"DBType"`
+		DataDirectoryPath string `yaml:"DataDirectoryPath"`
+		Addr              string `yaml:"Addr"`
+		Password          string `yaml:"Password"`
+		DB                int    `yaml:"DB"`
```

Comment:
Is it possible to only have `DBType` there and leave the rest of the config for DBs to figure out (that would probably mean another round of unmarshaling, but it's not a problem for me)?

Reply:
Take a look at d8bfcb9e80dec30ed33df753277a1afd3c16499f, it's quick and dirty, but I think that's the direction we should take.

Reply:
So you suggest to keep <Type>DB config structures keep within each DB instead of one place, right?

Reply:
Yep. Everything in `storage`, everything close to the specific DB that really uses these parameters. And specifies them. And at the same time it's all being avalable (and configurable) in the `config.Config`.

Reply:
If you're to add any parameter for WhateverDB -- you only touch `whateverdb_store.go`. If you're to add FancyNewDB --- you add `fancynewdb_store.go` and only one line to `store_config.go` (or maybe even move that type to `store.go`, one file for one type is probably too expensive).

Reply:
> (or maybe even move that type to store.go, one file for one type is probably too expensive
this one I believe won't work because of circular deps. That's why store_config located in config package.

Reply:
I mean to move `DBConfiguration` from `store_config.go` (where it's located after the d8bfcb9e80dec30ed33df753277a1afd3c16499f) to `store.go`, it's the same package, it'll work for sure.

Reply:
264a75dd5f8dc92fc81a7c1e479292b895aec0d9

Reply:
got it and done in the way